debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SourceInfo.java

```diff
@@ -380,6 +380,7 @@ public void startNextTransaction() {
      * Capture that we're starting a new event.
      */
     public void completeEvent() {
+        getNextId();
```

Comment:
not as currently done, see the below comment in regards to your changed API contract for a more complete answer, but essentially, the way SourceInfo is interacted with is always to first update the state and then build representations of that state for persisting to the offsetStore as well as the message itself.

In other words, we don't do anything with it, because we don't have anywhere to store it yet, that instead happens once we call `struct(TableId)` and `offset()`